packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -232,36 +246,50 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
                 onClick={!isCheckboxHidden ? this._onSelectAllClicked : undefined}
                 aria-colindex={1}
                 role={'columnheader'}
-                aria-hidden={isCheckboxHidden ? true : undefined}
               >
                 {onRenderColumnHeaderTooltip(
                   {
                     hostClassName: css(classNames.checkTooltip),
                     id: `${this._id}-checkTooltip`,
                     setAriaDescribedBy: false,
                     content: ariaLabelForSelectAllCheckbox,
-                    children: (
+                    children: !isCheckboxHidden ? (
                       <DetailsRowCheck
                         id={`${this._id}-check`}
                         aria-label={ariaLabelForSelectionColumn}
                         aria-describedby={
                           ariaLabelForSelectAllCheckbox && !this.props.onRenderColumnHeaderTooltip ? `${this._id}-checkTooltip` : undefined
                         }
-                        data-is-focusable={!isCheckboxHidden}
+                        data-is-focusable={true}
                         isHeader={true}
                         selected={isAllSelected}
                         anySelected={false}
-                        canSelect={!isCheckboxHidden}
+                        canSelect={true}
                         className={classNames.check}
                       />
+                    ) : (
+                      <div
+                        id={`${this._id}-selection`}
+                        className={css(rowCheckClassNames.root, rowCheckClassNames.check)}
+                        aria-label={ariaLabelForSelectionColumn}
+                        aria-describedby={
+                          ariaLabelForSelectionColumn && !this.props.onRenderColumnHeaderTooltip ? `${this._id}-checkTooltip` : undefined
+                        }
+                      />
                     )
                   },
                   this._onRenderColumnHeaderTooltip
                 )}
               </div>,
-              ariaLabelForSelectAllCheckbox && !this.props.onRenderColumnHeaderTooltip ? (

```

Comment:
All I did here was render `ariaLabelForSelectionColumn` in the label when `isCheckboxHidden` is true, since it becomes a header instead of a select all checkbox.